vars/githubCheck.groovy

```diff
@@ -0,0 +1,196 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+@Grab(group='io.jsonwebtoken', module='jjwt-impl', version='0.11.2')
+
+import io.jsonwebtoken.Jwts
+import java.security.interfaces.RSAPrivateKey
+import java.security.spec.PKCS8EncodedKeySpec
+import java.security.KeyFactory
+import java.util.Date
+import org.apache.commons.codec.binary.Base64
+import static io.jsonwebtoken.SignatureAlgorithm.RS256
+
+/**
+  Send GitHub check step
+
+  githubCheck(name: 'checkName', description: 'Execute something')
+
+*/
+def call(Map args = [:]) {
+  def name = args.containsKey('name') ? args.name : error('githubCheck: missing name argument')
+  def description = args.get('description', name)
+  def body = args.get('body', '')
+  def secret = args.get('secret', 'secret/observability-team/ci/github-app')
+  def org = args.get('org', env.ORG_NAME)
+  def repository = args.get('repository', env.REPO_NAME)
+  def commitId = args.get('commitId', env.GIT_BASE_COMMIT)
+  def status = args.get('status', 'neutral')
+  def detailsUrl = args.get('detailsUrl', '')
+
+  // Read secrets from vault
+  def props = getVaultSecret(secret: secret)
+  if(props?.errors){
+    error "Unable to get credentials from the vault: ${props.errors.toString()}"
+  }
+  def privateKeyContent = props?.data?.get('key')
+  def installationId = props?.data?.get('installation_id')
+  def appId = props?.data?.get('app_id')
+  // End read secrets from vault
+
+  // App login details
+  // TODO: potentially to be cached if within the valid timeframe.
+  def jsonWebToken = getJsonWebToken(privateKeyContent: privateKeyContent, appId: appId)
+  def token = getToken(jsonWebToken: jsonWebToken, installationId: installationId)
```

Comment:
NIT: I think we can split the step and provide a step to generate GitHub Tokens

Reply:
Just created `githubAppToken` -> https://github.com/elastic/apm-pipeline-library/pull/881/commits/708c200d2f561e0c2ef5494688173a1b8dce36aa